cmd/api/src/queries/graph.go

```diff
@@ -668,8 +668,9 @@ func (s *GraphQuery) CountFilteredNodes(ctx context.Context, filterCriteria grap
 	})
 }
 
-func (s *GraphQuery) CountNodesByKind(ctx context.Context, kinds ...graph.Kind) (int64, error) {
-	return s.CountFilteredNodes(ctx, (query.KindIn(query.Node(), kinds...)))
+func (s *GraphQuery) CountNodesByKind(ctx context.Context, filter []graph.Criteria, kinds ...graph.Kind) (int64, error) {
+	combinedFilter := query.And(filter, (query.KindIn(query.Node(), kinds...)))
+	return s.CountFilteredNodes(ctx, combinedFilter)
```

Comment:
Since filter is a slice, it should be exploded. Also consider renaming it to the plural filters since it is a slice. 
```suggestion
func (s *GraphQuery) CountNodesByKind(ctx context.Context, filter []graph.Criteria, kinds ...graph.Kind) (int64, error) {
    filter = append(filter, query.KindIn(query.Node(), kinds...))
	combinedFilter := query.And(filter...)
	return s.CountFilteredNodes(ctx, combinedFilter)
```